pyaedt/edb_core/edb_data/simulation_setup.py

```diff
@@ -0,0 +1,701 @@
+            "kSIwave": self._pedb.edb_api.utility.utility.SIWaveSimulationSetup,
+            "kLNA": None,
+            "kTransient": None,
+            "kQEye": None,
+            "kVEye": None,
+            "kAMI": None,
+            "kAnalysisOption": None,
+            "kSIwaveDCIR": self._pedb.edb_api.utility.utility.SIWaveDCIRSimulationSetup,
+            "kSIwaveEMI": None,
+            "kHFSSPI": None,
+            "kDDRwizard": None,
+            "kQ3D": None,
+            "kNumSetupTypes": None,
+        }
+        return setup_type_mapping[self.setup_type](self._edb_object)
+
+    @pyaedt_function_handler()
+    def _update_setup(self):
+        """Update setup into Edb."""
+        if self._setup_type == "kHFSS":
+            mesh_operations = self._edb_object.SimulationSettings.MeshOperations
+            mesh_operations.Clear()
+            for mop in self.mesh_operations.values():
+                mesh_operations.Add(mop.mesh_operation)
+
+        edb_setup = self._generate_edb_setup()
+        if self.name in self._pedb.setups:
+            self._pedb.layout.cell.DeleteSimulationSetup(self.name)
+        self._pedb.layout.cell.AddSimulationSetup(edb_setup)
+        return True
+
+
+class EdbFrequencySweep(object):
+    """Manages EDB methods for frequency sweep."""
+
+    def __init__(self, sim_setup, frequency_sweep=None, name=None, edb_sweep_data=None):
+        self._sim_setup = sim_setup
+
+        if edb_sweep_data:
+            self._edb_sweep_data = edb_sweep_data
+            self._name = self._edb_sweep_data.Name
+        else:
+            if not name:
+                self._name = generate_unique_name("sweep")
+            else:
+                self._name = name
+            self._edb_sweep_data = self._pedb.simsetupdata.SweepData(self._name)
+            self.set_frequencies(frequency_sweep)
+
+    @property
+    def _pedb(self):
+        """Edb."""
+        return self._sim_setup._pedb
+
+    @pyaedt_function_handler()
+    def _update_sweep(self):
+        """Update sweep."""
+        self._sim_setup._edb_object.SweepDataList.Clear()
+        for el in list(self._sim_setup.frequency_sweeps.values()):
+            self._sim_setup._edb_object.SweepDataList.Add(el._edb_sweep_data)
+        self._sim_setup._edb_object.SweepDataList.Add(self._edb_sweep_data)
+        return self._sim_setup._update_setup()
+
+    @property
+    def name(self):
+        """Name of the sweep."""
+        return self._edb_sweep_data.Name
+
+    @name.setter
+    def name(self, value):
+        """Set name of this sweep"""
+        self._edb_sweep_data.Name = value
+        self._update_sweep()
+
+    @property
+    def sweep_type(self):
+        """Sweep type."""
+        return
+
+    @property
+    def frequencies(self):
+        """List of frequencies points."""
+        return list(self._edb_sweep_data.Frequencies)
+
+    @property
+    def adaptive_sampling(self):
+        """Whether adaptive sampling is used.
```

Comment:
Properties are to be described as nouns. Here you are looking to see what has been set but not doing the setting. Properties don't have "Return" sections.

Reply:
That is correct @PipKat